File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task6.java
import java.util.Arrays;

public class Task6 {
    public static double findUniq(double[] arr) {
        if (arr == null || arr.length < 3) {
            throw new IllegalArgumentException("Array must contain at least 3 numbers.");
        }

        double a = arr[0], b = arr[1], c = arr[2];
        double majority;
        if (Double.compare(a, b) == 0 || Double.compare(a, c) == 0) {
            majority = a;
        } else {
            majority = b; // since a != b and a != c, b == c is the majority
        }

        for (double v : arr) {
            if (Double.compare(v, majority) != 0) {
                return v;
            }
        }

        // Should not reach here if input satisfies problem constraints
        throw new IllegalArgumentException("No unique element found.");
    }

    public static void main(String[] args) {
        double[][] tests = new double[][]{
            new double[]{1, 1, 1, 2, 1, 1},
            new double[]{0, 0, 0.55, 0, 0},
            new double[]{3, 4, 4, 4, 4},
            new double[]{-1, -1, -1, -2, -1, -1},
            new double[]{Double.NaN, Double.NaN, 7.0, Double.NaN}
        };

        for (double[] test : tests) {
            System.out.println(findUniq(test));
        }
    }
}

File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task6.py
import math

def find_uniq(arr):
    if arr is None or len(arr) < 3:
        raise ValueError("Array must contain at least 3 numbers.")

    a, b, c = arr[0], arr[1], arr[2]

    def eq(x, y):
        return (math.isnan(x) and math.isnan(y)) or x == y

    if eq(a, b) or eq(a, c):
        majority = a
    else:
        majority = b  # since a != b and a != c, b == c is the majority

    for v in arr:
        if not eq(v, majority):
            return v

    raise ValueError("No unique element found.")

if __name__ == "__main__":
    tests = [
        [1, 1, 1, 2, 1, 1],
        [0, 0, 0.55, 0, 0],
        [3, 4, 4, 4, 4],
        [-1, -1, -1, -2, -1, -1],
        [float('nan'), float('nan'), 7.0, float('nan')],
    ]
    for t in tests:
        print(find_uniq(t))